### PR TITLE
reuse persistent temp dir where needed

### PR DESCRIPTION
**Pull request by dgdonut**

### CODE REVIEW

File: binaries/scoot-snapshot-db/main.go [hunk 1]
```diff
@@ -53,6 +56,11 @@ func (i *injector) RegisterFlags(rootCmd *cobra.Command) {
 }
 
 func (i *injector) Inject() (snapshot.DB, error) {
+	dbTempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (ryancouto) [inline binaries/scoot-snapshot-db/main.go:61]: not part of this PR, but could we remove the comment below (or at least the parts about talking to the daemon)

File: binaries/scoot-snapshot-db/main.go [hunk 2]
```diff
@@ -36,11 +37,13 @@ func main() {
 
 	inj := &injector{}
 	cmd := cli.MakeDBCLI(inj)
-	if err := cmd.Execute(); err != nil {
-		removeTemp()
+	err = cmd.Execute()
+	if dbTempDir != "" {
+		os.RemoveAll(dbTempDir)
```
Comment (ryancouto) [inline binaries/scoot-snapshot-db/main.go]: when will dbTempDir not be ""?

File: snapshot/git/gitdb/setup.go [hunk 3]
```diff
@@ -38,6 +38,9 @@ func (m module) Install(b *ice.MagicBag) {
 		func() AutoUploadDest {
 			return AutoUploadBundlestore
 		},
+		func() string {
```
Comment (ryancouto) [inline snapshot/git/gitdb/setup.go:41]: honestly don't really understand why we need this... is this dbTempDir?

File: binaries/scoot-snapshot-db/main.go (hunk 2 above)
Comment (dgdonut) [reply binaries/scoot-snapshot-db/main.go]: If tempdir creation fails. Otherwise it will do os.RemoveAll("") - which I'm not sure exactly what that would do. Nothing? Or delete everything in your cwd?

File: snapshot/git/gitdb/setup.go (hunk 3 above)
Comment (dgdonut) [reply snapshot/git/gitdb/setup.go:41]: oss apiserver main.go implicitly calls MakeDBNewRepo here via ice, and that func needs a string for (a preexisting) tmpdir path. In closed source we call it explicitly, without ice

File: binaries/scoot-snapshot-db/main.go (hunk 2 above)
Comment (dgdonut) [reply binaries/scoot-snapshot-db/main.go]: checked the os pkg implementation and "" is special cased to do nothing and return nil, so removing the check.

File: snapshot/git/gitdb/setup.go (hunk 3 above)
Comment (ryancouto) [reply snapshot/git/gitdb/setup.go:41]: Does that mean that the oss MakeDBNewRepo will use / as temp then? Should we do os.TempDir() here instead of ""?

Comment (dgdonut) [reply snapshot/git/gitdb/setup.go:41]: It will effectively pass "" through which will result in ioutil.TempDir("", ...) calls, which is equivalent to using os.TempDir()

Comment (dgdonut) [reply snapshot/git/gitdb/setup.go:41]: ex:

> $ go run github.com/twitter/scoot/apiserver
INFO[0000] Making new FileStore at dir: /var/folders/y0/6frrxtkn495dgbzr_s_7zcgr0000gp/T/bundles074177289  sourceLine="store/file_store.go:26"
INFO[0000] Starting new bundlestore.Server with root: /var/folders/y0/6frrxtkn495dgbzr_s_7zcgr0000gp/T/bundles074177289  sourceLine="bundlestore/server.go:34"
INFO[0000] git.NewRepository: /var/folders/y0/6frrxtkn495dgbzr_s_7zcgr0000gp/T/gitdb-repo-541018836, top: /private/var/folders/y0/6frrxtkn495dgbzr_s_7zcgr0000gp/T/gitdb-repo-541018836  sourceLine="repo/repo.go:132"

**Review by ryancouto — APPROVED** [review]
